src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -3057,7 +3068,13 @@ internal static Attribute GetCustomAttributeHelper(Type type, Type attributeType
 
         // Helper to deal with the fact that the type we are reflecting over might be loaded in the ReflectionOnly context.
         // When that is the case, we have the build the custom assemblies on a member by hand.
-        internal static Attribute? GetCustomAttributeHelper(MemberInfo member, Type attributeType, EventManifestOptions flags = EventManifestOptions.None)
+        internal static Attribute? GetCustomAttributeHelper(
+            MemberInfo member,
+#if !ES_BUILD_STANDALONE
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.PublicProperties)]
```

Comment:
I was thinking of just suppressing these two messages. The reasoning being:

1. We are using `CustomAttributeData.GetCustomAttributes` to read the attribute information.
2. We are only calling Activator.CreateInstance on an attribute if someone else is already using that attribute's constructor.
3. We are only setting the attributes' properties if someone else is already setting that property.

In these cases, the constructor and setter will already be preserved, since someone else (the place declaring the attribute) is already creating the attribute and setting its properties.

Thoughts?

Reply:
Doesn't this activate a different attribute though? The way I read this is that this is doing a name match with the `attributeType` and if the name of the attribute that is actually applied matches the `attributeType`, it will activate `attributeType`. The matching attribute could be `attributeType` exactly, but it could also be a different type loaded in the ReflectionOnly load context (that we couldn't activate even if we wanted to).

Reply:
Yep you are right. Even further, it might not be ReflectionOnly context, but there is an OOB package with a different namespace: `Microsoft.Diagnostics.Tracing`, and the `AttributeTypeNamesMatch` method will match the attributes from the `Microsoft` namespace to attributes in the `System.Diagnostics.Tracing` namespace.